src/persist-client/src/lib.rs

```diff
@@ -45,8 +45,10 @@ pub mod batch;
 pub mod cache;
 pub mod error;
 pub mod read;
+pub mod usage;
 pub mod write;
 
+pub use crate::r#impl::paths::BlobKeyPrefix;
```

Comment:
I personally try to stay away from `pub use` (the Since, Upper one below is my doing and I already regret it). I think this might be a decent argument for either `mod paths` being pub (and moved out of impl) or switching StorageUsageClient back to taking a ShardId. maybe the latter?

Reply:
oh interesting, I thought the targeted `pub` here was a nice balance but I also have never dealt with Rust module structure before :) 

I'll poke at this and see what makes sense. Maybe we can make the current `size` fn private and offer a public `shard_size` fn for external consumption

Reply:
in the interest of transparency, this is very much a matter of taste. I dislike the `pub use` because within the crate impl itself, it creates ambiguity on which one you should `use`

Reply:
merged this feedback together with the retry_external comment, and restored the `pub shard_size` fn to internally retry the now-private `size` fn